nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/test/groovy/org/apache/nifi/NiFiGroovyTest.groovy

```diff
@@ -105,14 +97,13 @@ class NiFiGroovyTest extends GroovyTestCase {
         NiFi.main(args)
 
         // Assert
-        assert TestAppender.events.last().getMessage() == "Failure to launch NiFi due to java.lang.IllegalArgumentException: The bootstrap process did not provide a valid key and there are protected properties present in the properties file"
+        assert TestAppender.events.last().getMessage() == "Failure to launch NiFi due to java.lang.IllegalArgumentException: There was an issue decrypting protected properties"
```

Comment:
This error message seems less helpful than the previous message. Is there a reason it is changed?

Reply:
Some of the exception handling was changed in Nov 2016, after which that message was not returned, and instead a shortened version of it is used.  The commit of that change is https://github.com/apache/nifi/commit/3e892c55ec35bb47c67e6a5d6d16e168f1b2f0fe#diff-4f0202fa752f36587f9eaade42fb7cb5R280.  Where that exception was thrown, though, I don't think it could yet have determined that there are encrypted properties, only that the key was not formatted correctly, or the JCE Unliminited Strength Jurisdiction Policies aren't available.